pallets/oracle-data-collection/src/benchmarking.rs

```diff
@@ -0,0 +1,153 @@
+use cfg_traits::{benchmarking::PoolBenchmarkHelper, changes::ChangeGuard, ValueProvider};
+use frame_benchmarking::{v2::*, whitelisted_caller};
+use frame_support::storage::bounded_vec::BoundedVec;
+use frame_system::RawOrigin;
+
+use crate::{
+	pallet::{Call, Collection, Config, Pallet},
+	types::Change,
+};
+
+#[cfg(test)]
+fn init_mocks() {
+	use crate::mock::{MockChangeGuard, MockIsAdmin, MockProvider};
+
+	MockIsAdmin::mock_check(|_| true);
+	MockProvider::mock_get(|_, _| Ok(Some((Default::default(), Default::default()))));
+	MockChangeGuard::mock_note(|_, change| {
+		MockChangeGuard::mock_released(move |_, _| Ok(change.clone()));
+		Ok(Default::default())
+	});
+}
+
+mod util {
+	use super::*;
+
+	pub fn last_change_id_for<T>(
+		key: T::OracleKey,
+		feeders: &BoundedVec<T::FeederId, T::MaxFeedersPerKey>,
+	) -> T::Hash
+	where
+		T: Config,
+		T::CollectionId: Default,
+	{
+		// Emulate to note a change to later apply it
+		T::ChangeGuard::note(
+			T::CollectionId::default(),
+			Change::<T>::Feeders(key, feeders.clone()).into(),
+		)
+		.unwrap()
+	}
+}
+
+#[benchmarks(
+    where
+        T::CollectionId: Default,
+        T::OracleKey: Default + From<u32>,
+        T::OracleValue: Default,
+        T::Timestamp: Default,
+        T::Hash: Default,
+        T::FeederId: From<T::AccountId>,
+        T::ChangeGuard: PoolBenchmarkHelper<PoolId = T::CollectionId, AccountId = T::AccountId>,
+    )]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
+	fn propose_update_feeders(n: Linear<1, 10>) -> Result<(), BenchmarkError> {
+		#[cfg(test)]
+		init_mocks();
+
+		let admin: T::AccountId = whitelisted_caller();
+
+		T::ChangeGuard::bench_create_pool(T::CollectionId::default(), &admin);
+
+		let feeders = (0..n)
+			.map(|i| account::<T::AccountId>("feeder", i, 0).into())
+			.collect::<Vec<_>>()
+			.try_into()
+			.unwrap();
+
+		#[extrinsic_call]
+		propose_update_feeders(
+			RawOrigin::Signed(admin),
+			T::CollectionId::default(),
+			T::OracleKey::default(),
+			feeders,
+		);
+
+		Ok(())
+	}
+
+	#[benchmark]
+	fn apply_update_feeders(n: Linear<1, 10>) -> Result<(), BenchmarkError> {
```

Comment:
Yes, you can do it. But I thought that if some runtime configures a really high value, the benchmark could take a lot of time to compute (especially in the other benchmark that has 2 variables). Because it's linear, `10` values should be enough to determine the correct linear function for later infer high values